src/rules/disallow-references.ts

```diff
@@ -0,0 +1,47 @@
+import { createRule } from '../utils/create-rule';
+
+type MessageIds = 'noAssignRef' | 'noAssignRefFix';
+type Options = [];
+
+export const disallowReferences = createRule<MessageIds, Options>({
+	meta: {
+		type: 'layout',
```

Comment:
```suggestion
		type: 'suggestion',
```

(we need to fix others as well)